HighwayPatrol.cpp

```diff
@@ -10,7 +11,7 @@ void HighwayPatrol::scanHighway(Highway* h)
 {
     std::cout << name << ": scanning highway for speeders" << std::endl;
 
-    for( size_t i = h->vehicles.size(); --i >= 0; )
+    for( size_t i = 0; i < h->vehicles.size(); ++i  )
```

Comment:
This causes undefined behavior because `h->removeVehicle()` erases objects from the vector. That's why it loops from last to first, so that any deleted elements are not going to be iterated over.   You added a bug when you revised which way the loop runs.
- [ ] revise.